src/commands/remoteDebug/startRemoteSsh.ts

```diff
@@ -0,0 +1,110 @@
+
+        // remote debugging has to be disabled in order to tunnel to the 2222 port
+        await remoteDebug.setRemoteDebug(false, undefined /*skips confirmation*/, undefined, siteClient, siteConfig);
+
+        remoteDebug.reportMessage('Starting tunnel proxy...', progress);
+
+        const publishCredential: User = await siteClient.getWebAppPublishCredential();
+        const tunnelProxy: TunnelProxy = new TunnelProxy(portNumber, siteClient, publishCredential);
+        await callWithTelemetryAndErrorHandling('appService.remoteSshStartProxy', async function (this: IActionContext): Promise<void> {
+            this.rethrowError = true;
+            await tunnelProxy.startProxy(2222);
+            await connectToTunnelProxy(tunnelProxy);
+        });
+    });
+
+    async function connectToTunnelProxy(tunnelProxy: TunnelProxy): Promise<void> {
+        const sshTerminalName: string = `${node.root.client.fullName} - Remote SSH`;
+        // -o StrictHostKeyChecking=no doesn't prompt for adding to hosts
+        // -o "UserKnownHostsFile /dev/null" doesn't add host to known_user file
+        // -o "LogLevel ERROR" doesn't display Warning: Permanently added 'hostname,ip' (RSA) to the list of known hosts.
+        const sshCommand: string = `ssh -c aes256-cbc -o StrictHostKeyChecking=no -o "UserKnownHostsFile /dev/null" -o "LogLevel ERROR" root@127.0.0.1 -p ${portNumber}`;
+        const terminal: vscode.Terminal = vscode.window.createTerminal(sshTerminalName);
+
+        // because the container needs time to respond, there needs to be a delay between connecting and entering password
+        terminal.sendText(sshCommand, true);
+        await delay(3000);
+        terminal.sendText('Docker!', true);
+        terminal.show();
+        ext.context.subscriptions.push(terminal);
+
+        vscode.window.onDidCloseTerminal(async (e: vscode.Terminal) => {
+            if (e.processId === terminal.processId) {
+                // clean up if the SSH task ends
+                if (tunnelProxy !== undefined) {
+                    tunnelProxy.dispose();
+                }
+                remoteSsh.set(node.root.client.fullName, false);
+                ext.outputChannel.appendLine(`Azure Remote SSH for "${node.root.client.fullName}" has disconnected.`);
+                await remoteDebug.setRemoteDebug(oldSetting, undefined/*skips confirmation*/, undefined, siteClient, siteConfig);
+            }
+        });
+    }
+}
+
+export async function stopRemoteSsh(node?: SiteTreeItem): Promise<void> {
+    if (!node) {
+        node = <SiteTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+    }
+
+    if (!remoteSsh.get(node.root.client.fullName)) {
+        throw new Error(`Azure Remote SSH is not currently running for "${node.root.client.fullName}".`);
+    }
+    const sshTerminalName: string = `${node.root.client.fullName} - Remote SSH`;
+    for (const terminal of vscode.window.terminals) {
```

Comment:
Your `remoteSsh` map above should somehow contain the terminal so that you don't have to search for it by name